gui/autounsuspend.lua

```diff
@@ -0,0 +1,77 @@
+-- config ui for autounsuspend
+
+local gui = require("gui")
+local widgets = require("gui.widgets")
+local autounsuspend = reqscript("autounsuspend")
+
+local REFRESH_MS = 10000
+
+---
+-- Autounsuspend
+---
+
+Autounsuspend = defclass(Autounsuspend, widgets.Window)
+Autounsuspend.ATTRS{
+    frame_title = "Autounsuspend",
+    frame = {w=35, h=11},
+    resizable = false
+}
+
+function Autounsuspend:init()
+    self:addviews{
+        widgets.ToggleHotkeyLabel{
+            view_id="enable_toggle",
+            frame={t=0, l=0, w=31},
+            label="Autounsuspend is",
+            key="CUSTOM_CTRL_E",
+            options={{value=true, label="Enabled", pen=COLOR_GREEN},
+                     {value=false, label="Disabled", pen=COLOR_RED}},
+            on_change=function(val) dfhack.run_command{val and "enable" or "disable", "autounsuspend"} end
+        },
+        widgets.ToggleHotkeyLabel{
+            view_id="smartMode",
+            frame={t=3, l=0, w=31},
+            label="Smart suspend:",
+            key="CUSTOM_ALT_S",
+            options={{value=true, label="Yes", pen=COLOR_GREEN},
+                     {value=false, label="No", pen=COLOR_RED}},
+            on_change=function(val)
+                autounsuspend.set_smart_mode(val)
+            end
+        },
+
+    }
+    self:refresh_data()
+end
+
+function Autounsuspend:refresh_data()
+    self.subviews.enable_toggle:setOption(autounsuspend.isEnabled())
+    self.subviews.smartMode:setOption(autounsuspend.smart)
+    self.next_refresh_ms = dfhack.getTickCount() + REFRESH_MS
+end
+
+function Autounsuspend:onRenderBody()
+    if self.next_refresh_ms <= dfhack.getTickCount() then
+        self:refresh_data()
+    end
```

Comment:
unless you're reporting on the live number of suspended jobs, there's really no reason to have an autorefresh.

Reply:
oops forgot to adress this one

Reply:
And done this time